src/MonitorProfiler/DllMain.cpp

```diff
@@ -16,18 +18,14 @@ STDMETHODIMP_(BOOL) DLLEXPORT DllMain(HMODULE hModule, DWORD ul_reason_for_call,
 
 STDAPI DLLEXPORT DllGetClassObject(REFCLSID rclsid, REFIID riid, LPVOID* ppv)
 {
-    if (nullptr == ppv)
-    {
-        return E_POINTER;
-    }
-
-    ComPtr<ClassFactory> pfactory(new ClassFactory(rclsid));
-    if (nullptr == pfactory)
-    {
-        return E_OUTOFMEMORY;
-    }
-
-    return pfactory->QueryInterface(riid, ppv);
+    IfNullPointer(ppv);
+
+    *ppv = nullptr;
```

Comment:
Just a sanity check that we intend to set this to `nullptr` here - in the code being replaced we don't do this right now, so I wanted to make sure that this change was intentional.

Reply:
These were intentional. I should have added these in the initial merge but missed it.

We should be initializing double pointers to a reasonable value (typically nullptr) before doing anything so that if we return from the method early then the points are already initialized with appropriate values.